src/Game/GameObject/GameObject.js

```diff
@@ -353,6 +353,58 @@ const GameObjectModule = class GameObject {
     return this.computeForwardVector().applyQuaternion(quaternion);
   }
 
+  /**
+   * compute Upward vector
+   * @returns {THREE.Vector3}
+   */
+  computeUpVector() {
+    const quaternion = new THREE.Quaternion().setFromEuler(
+      this.object3D.rotation
+    );
+    // console.log('coucou UP');
+    const result = new THREE.Vector3(0, 0, 1).applyQuaternion(quaternion);
+    return result;
+  }
+
+  /**
+   * Compute the downward vector 
+   * @returns {THREE.Vector3}
+   */
+  computeDownVector() {
+    const quaternion = new THREE.Quaternion().setFromEuler(
+      this.object3D.rotation
+    );
+    // console.log('coucou DOWN');
+    const result = new THREE.Vector3(0, 0, -1).applyQuaternion(quaternion);
+    return result;
+  }
+
+  /**
+   * Compute the left translation vector 
+   * @returns {THREE.Vector3}
+   */
+  computeLeftVector() {
+    const quaternion = new THREE.Quaternion().setFromEuler(
+      this.object3D.rotation
+    );
+    //console.log('coucou LEFT');
+    const result = new THREE.Vector3(-1, 0, 0).applyQuaternion(quaternion);
+    return result;
+  }
+
+  /**
+   * Compute the right translation vector 
+   * @returns {THREE.Vector3}
+   */
+   computeRightVector() {
+    const quaternion = new THREE.Quaternion().setFromEuler(
+      this.object3D.rotation
+    );
+    //console.log('coucou RIGHT');
+    const result = new THREE.Vector3(1, 0, 0).applyQuaternion(quaternion);
```

Comment:
compute new THREE.Vector(1,0,0) from defaultForward vector of the gameobject class